src/App.tsx

```diff
@@ -14,11 +16,104 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isVisible: boolean,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [...goodsFromServer],
+    isVisible: true,
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.localeCompare(b)),
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+    });
+  };
+
+  sortByLength = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  render() {
+    const { goods, isVisible } = this.state;
+
+    return (
+      <div className="app">
+        {
+          isVisible
+            ? (
```

Comment:
Maybe, it'll be better to use conditional rendering instead of render using ternary operator.